engine/src/main/scala/cromwell/jobstore/SqlJobStore.scala

```diff
@@ -0,0 +1,60 @@
+package cromwell.jobstore
+
+import cromwell.core.{JobOutputs, WorkflowId}
+import cromwell.database.sql.JobStoreSqlDatabase
+import cromwell.database.sql.tables.JobStoreEntry
+
+import scala.concurrent.{ExecutionContext, Future}
+import cromwell.core.ExecutionIndex._
+import spray.json._
+
+class SqlJobStore(sqlDatabase: JobStoreSqlDatabase) extends JobStore {
+  override def writeToDatabase(jobCompletions: Map[JobStoreKey, JobResult], workflowCompletions: List[WorkflowId])(implicit ec: ExecutionContext): Future[Unit] = {
+    for {
+      _ <- sqlDatabase.addJobStoreEntries(jobCompletions map toJobStoreEntry)
+      _ <- sqlDatabase.removeJobResultsForWorkflows(workflowCompletions map {_.toString})
+    } yield ()
+  }
+
+  private def toJobStoreEntry(jobCompletion: (JobStoreKey, JobResult)): JobStoreEntry = {
+    import JobResultJsonFormatter._
+    jobCompletion match {
+      case (key, JobResultSuccess(returnCode, jobResult)) => JobStoreEntry(
+        key.workflowId.toString,
+        key.callFqn,
+        key.index.fromIndex,
+        key.attempt,
+        jobSuccessful = true,
+        returnCode,
+        Some(jobResult.toJson.toString),
+        None,
+        None)
+      case (key, JobResultFailure(returnCode, throwable, retryable)) => JobStoreEntry(
+        key.workflowId.toString,
+        key.callFqn,
+        key.index.fromIndex,
+        key.attempt,
+        jobSuccessful = true,
```

Comment:
interesting that you'd point out this one but not comment that the unsuccessful case also still had `successful = true`...